### PR TITLE
Enable some debug info tests

### PR DESCRIPTION
**Pull request by EFanZh**
Try to fix and enable some debug info tests, this should help avoid some accidental breakages.

### CODE REVIEW

File: src/test/debuginfo/type-names.rs [hunk 1]
```diff
@@ -36,12 +35,12 @@
...
-// gdbg-check:type = union Enum3<type_names::mod1::Struct2>
-// gdbr-check:type = type_names::mod1::mod2::Enum3<type_names::mod1::Struct2>
+// gdbg-check:type = union Enum3
+// gdbr-check:type = type_names::mod1::mod2::Enum3
 
 // gdb-command:whatis generic_enum_2
-// gdbg-check:type = union Enum3<type_names::Struct1>
-// gdbr-check:type = type_names::mod1::mod2::Enum3<type_names::Struct1>
+// gdbg-check:type = union Enum3
+// gdbr-check:type = type_names::mod1::mod2::Enum3
```
Comment (EFanZh) [inline src/test/debuginfo/type-names.rs:43]: Are these type name changes expected?

File: src/test/debuginfo/option-like-enum.rs [hunk 2]
```diff
@@ -22,19 +22,21 @@
 
 // gdbg-command:print empty_gdb->discr
 // gdbr-command:print empty_gdb.discr
-// gdb-check:$4 = (isize *) 0x0
+// gdbg-check:$4 = (isize *) 0x1
+// gdbr-check:$4 = (*mut isize) 0x1
```
Comment (EFanZh) [inline src/test/debuginfo/option-like-enum.rs:26]: Is the value change expected?

**Comment by EFanZh** [comment]
There are some behavior changes that I am not sure whether they are expected. I have added comments to those changes.

**Comment by Mark-Simulacrum** [comment]
I'm going to go ahead and give this a close for now, but feel free to reopen (or open a new PR) when you'd like to get it reviewed.

**Pull request closed without merging**